engine/flink/management/src/main/scala/pl/touk/nussknacker/engine/management/FlinkDeploymentManager.scala

```diff
@@ -232,12 +234,16 @@ class FlinkDeploymentManager(
           .sequence(
             deploymentIdsToCheck.toSeq
               .map { deploymentId =>
-                client
-                  .getJobDetails(deploymentId.toJobID)
-                  .map(_.map { jobDetails =>
-                    deploymentId -> FlinkStatusDetailsDeterminer
-                      .toDeploymentStatus(JobStatus.valueOf(jobDetails.state), jobDetails.`status-counts`)
-                  })
+                (for {
+                  jobDetails <- OptionT(client.getJobDetails(deploymentId.toJobID))
+                  jobConfig  <- OptionT.liftF(client.getJobConfig(deploymentId.toJobID))
+                } yield deploymentId -> FlinkStatusDetailsDeterminer
+                  .toDeploymentStatus(
+                    JobStatus.valueOf(jobDetails.state),
+                    jobDetails.`status-counts`,
+                    Instant.ofEpochMilli(jobDetails.`start-time`),
+                    jobConfig.versionId.getOrElse(VersionId(0))
```

Comment:
Let's pass `None` to the place where we can decide if it is a situation for either IllegalStateException or some fallback with warning log